packages/m2m-gateway/src/services/eserviceTemplateService.ts

```diff
@@ -90,5 +110,303 @@ export function eserviceTemplateServiceBuilder(
+          seed,
+          {
+            params: { templateId },
+            headers,
+          }
+        );
+      const polledResource = await pollEServiceTemplate(response, headers);
+      return toM2MGatewayEServiceTemplate(polledResource.data);
+    },
+    async createEServiceTemplateVersion(
+      templateId: EServiceTemplateId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplateVersion> {
+      logger.info(
+        `Creating new version for eservice template with id ${templateId}`
+      );
+
+      const createdVersion =
+        await clients.eserviceTemplateProcessClient.createEServiceTemplateVersion(
+          undefined,
+          {
+            params: { templateId },
+            headers,
+          }
+        );
+
+      // Poll the template to ensure the version was created
+      const polledTemplate = await pollEServiceTemplate(
+        {
+          ...(await retrieveEServiceTemplateById(headers, templateId)),
+          metadata: createdVersion.metadata,
+        },
+        headers
+      );
+
+      const version = polledTemplate.data.versions.find(
+        (v) => v.id === createdVersion.data.id
+      );
+      if (!version) {
+        throw eserviceTemplateVersionNotFound(
+          templateId,
+          unsafeBrandId(createdVersion.data.id)
+        );
+      }
+      return toM2MGatewayEServiceTemplateVersion(version);
+    },
+    async updateDraftEServiceTemplateVersion(
+      templateId: EServiceTemplateId,
+      versionId: EServiceTemplateVersionId,
+      seed: {
+        description?: string;
+        voucherLifespan?: number;
+        dailyCallsPerConsumer?: number;
+        dailyCallsTotal?: number;
+        agreementApprovalPolicy?: "AUTOMATIC" | "MANUAL";
+      },
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplateVersion> {
+      logger.info(
+        `Updating version ${versionId} of eservice template with id ${templateId}`
+      );
+
+      const response =
+        await clients.eserviceTemplateProcessClient.patchUpdateDraftTemplateVersion(
+          seed,
+          {
+            params: { templateId, templateVersionId: versionId },
+            headers,
+          }
+        );
+      const polledResource = await pollEServiceTemplate(response, headers);
+
+      const version = polledResource.data.versions.find(
+        (v) => v.id === versionId
+      );
+      if (!version) {
+        throw eserviceTemplateVersionNotFound(templateId, versionId);
+      }
+      return toM2MGatewayEServiceTemplateVersion(version);
+    },
+    async deleteDraftEServiceTemplateVersion(
+      templateId: EServiceTemplateId,
+      versionId: EServiceTemplateVersionId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<void> {
+      logger.info(
+        `Deleting version ${versionId} for eservice template with id ${templateId}`
+      );
+
```

Comment:
See what we did in deleteDraftEServiceDescriptor - we must check that we are not deleting the last version, otherwise we are deleting the entire template and we don't want to expose two different functionalities in a single endpoint. Then we will consider adding also the method to delete the template (added it as a point to the next release)